src/DatatypeChannels.ASB/Consumer.fs

```diff
@@ -37,18 +42,21 @@ let mkNew (options: Options)
     let ctxRef = ref Option<ServiceBusReceiver * (unit -> Activity) * ConcurrentDictionary<string, MessageContext>>.None
     let activitySource = Diagnostics.mkActivitySource "Consumer"
     let withCtx cont =
+        let startCtx msgs (name:Task<_>) =
+            backgroundTask {
+                use activity = activitySource |> Diagnostics.startActivity (Diagnostics.formatName "NewConsumerCtx") ActivityKind.Consumer
+                let! name = name
+                activity.AddTag("binding", $"{name}/{options.SubQueue}") |> ignore
+                let startActivity _ = activitySource |> Diagnostics.startActivity name ActivityKind.Consumer
+                let receiver = withClient (fun client -> client.CreateReceiver(name, options))
+                ctxRef.Value <- Some (receiver, startActivity, msgs)
+                return ctxRef.Value.Value
+            }
         match ctxRef.Value with
-        | Some ((receiver,_,_) as ctx) when not receiver.IsClosed -> Task.FromResult ctx
-        | _ -> fun (name:Task<_>) -> backgroundTask {
-                    use activity = activitySource |> Diagnostics.startActivity "new" ActivityKind.Consumer
```

Comment:
ActivitySourceName doesn't show up in geneva, so have to be verbose in naming them.